--- app/Http/Middleware/TrustProxies.php.orig
+++ app/Http/Middleware/TrustProxies.php
@@ -13,6 +13,7 @@
 
 use Fideloper\Proxy\TrustProxies as Middleware;
 use Illuminate\Http\Request;
+use Illuminate\Contracts\Config\Repository;
 
 class TrustProxies extends Middleware
 {
@@ -29,4 +30,15 @@
      * @var int
      */
     protected $headers = Request::HEADER_X_FORWARDED_ALL;
+
+    /**
+     * Instantiate new trusted proxies middleware
+     *
+     * @param \Illuminate\Contracts\Config\Repository $config
+     */
+    public function __construct(Repository $config) {
+	    parent::__config($config);
+	    if (config('ninja.trusted_proxies'))
+		    $this->proxies = config('ninja.trusted_proxies');
+    }
 }
--- config/ninja.php.orig
+++ config/ninja.php
@@ -29,6 +29,7 @@
     'enabled_modules' => 32767,
     'phantomjs_key' => env('PHANTOMJS_KEY', false),
     'phantomjs_secret' => env('PHANTOMJS_SECRET', false),
+    'trusted_proxies' => env('TRUSTED_PROXIES', false),
 
     'sentry_dsn' => env('SENTRY_LARAVEL_DSN', 'https://9b4e15e575214354a7d666489783904a@sentry.invoicing.co/6'),
     'environment' => env('NINJA_ENVIRONMENT', 'selfhost'), // 'hosted', 'development', 'selfhost', 'reseller'
--- .env.example.orig
+++ .env.example
@@ -48,6 +48,7 @@
 GOOGLE_MAPS_API_KEY=
 API_SECRET=superdoopersecrethere
 ERROR_EMAIL=
+TRUSTED_PROXIES=
 
 NINJA_ENVIRONMENT=selfhost
 
